fl4health/reporting/base_reporter.py

```diff
@@ -0,0 +1,51 @@
+"""Base Class for Reporters.
+
+Super simple for now but keeping it in a seperate file in case we ass more base methods.
+"""
+
+from typing import Any
+
+
+class BaseReporter:
+    def report(
+        self,
+        data: dict,
+        round: int | None = None,
+        epoch: int | None = None,
+        batch: int | None = None,
+    ) -> None:
+        """A method called by clients or servers to send data to the reporter.
+
+        The report method is called reduncantly by the clients every step, every epoch,
+        every round and sometimes outside of a FL round (for high level summary data).
+        It is up to the reporter to determine when and what to report.
+
+        Args:
+            data (dict): The data to maybe report from the server or client.
+            round (int | None, optional): The current FL round. If None, this indicates
+                that the method was called outside of a round (eg. for summary
+                information). Defaults to None.
+            epoch (int | None, optional): The current epoch. If None then this method
+                was not called within the scope of an epoch. Defaults to None.
+            batch (int | None, optional): The current batch. If None then this method
+                was called outside the scope of a training or evaluation step (eg. at
+                the end of an epoch or round) Defaults to None.
+        """
+        raise NotImplementedError
+
+    def initialize(self, **kwargs: Any) -> None:
```

Comment:
Should these last two methods raise NotImplementedError or are we assuming by default they do nothing unless overridden?

Reply:
My intention was to assume they do nothing unless overridden. In most cases I felt that they would not be needed and didn't want to force subclasses to override them just to pass